tools/topology/topology2/include/bench/bench_comp_generate.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+set -e
+
+main () {
+    if [ "$#" -ne 1 ]; then
+	echo "Usage: $0 <component>"
+	exit 1
+    fi
+    comp=$1
+
+    generate_comp "s16"
+    generate_comp "s24"
+    generate_comp "s32"
+    generate_route
+    generate_playback_controls
+    generate_capture_controls
+}
+
+generate_comp ()
+{
+    format=$1
+    fn=${comp}_${format}.conf
+    echo Creating file "$fn"
+    cat > "$fn" <<EOF_COMP
+		# Created with script "$0 $comp"
```

Comment:
```suggestion
		# Created with script ${FULL_CMD[@]}
```

This addresses @lgirdwood 's concern.

It does not "re-quote" whitespace but good enough for now.

Reply:
Thanks, yes that's better, though it's not same as what Liam asked.